### PR TITLE
FreeBSD: Cleanup zfs_readdir()

### PR DESCRIPTION
**Pull request by ryao**
### Motivation and Context
The FreeBSD project's coverity scans found dead code in `zfs_readdir()`. Also, there is a dead label and the comment above `zfs_readdir()` is out of date.

### Description
I fixed the comment and deleted all of the dead code.

### How Has This Been Tested?
The buildbot could test.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by freqlabs** [comment]
This makes it clear we can also get rid of the definitions for `V_RDDIR_*` flags, the `eodp` local in `zfs_readdir()`, and `extdirent.h` entirely. While in `include/os/freebsd/spl/sys/vnode_impl.h`, we don't need the `V_XATTRDIR` or `V_ACE_MASK` flags defined there either.

**Comment by ryao** [comment]
@freqlabs Nice find. I just pushed a revision that deletes all of the additional things you mentioned.

**Review by freqlabs — APPROVED** [review]
Thank you!

File: include/os/linux/spl/sys/file.h [hunk 1]
```diff
@@ -26,7 +26,6 @@
 
 #define	FIGNORECASE		0x00080000
 #define	FKIOCTL			0x80000000
-#define	ED_CASE_CONFLICT	0x10
```
Comment (freqlabs) [inline include/os/linux/spl/sys/file.h:29]: Whoops, looks like this one is still used.

Comment (ryao) [reply include/os/linux/spl/sys/file.h]: I just put that back. Thanks for catching it so quickly.

**Review by behlendorf — APPROVED** [review]

**Pull request merged**